src/api/goods.ts

```
@@ -3,14 +3,43 @@
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+export async function getAll(): Promise<Good[]> {
+  try {
+    const response = await fetch(API_URL);
+
+    if (!response.ok) {
+      throw new Error(`error: ${response.status} ${response.statusText}`);
+    }
+
+    const data: Good[] = await response.json();
+
+    return data;
+  } catch (error) {
+    console.error('error getAll', error);
+    throw error;
+  }
 }
 
-export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+export const get5First = async (): Promise<Good[]> => {
+  try {
+    const goods = await getAll();
+
+    return [...goods].sort((a, b) => a.name.localeCompare(b.name)).slice(0, 5);
+  } catch (err) {
+    console.error('error get5First', err);
+
+    return [];
+  }
 };
 
-export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+export const getRed = async (): Promise<Good[]> => {
+  try {
+    const goods = await getAll();
+
+    return goods.filter(good => good.color === 'red');
+  } catch (err) {
+    console.error('error getRed', err);
+
+    return [];
+  }
```

Similar to `get5First`, this `catch` block prevents the error from reaching the `App` component's error handler. The promise from `getRed` will always resolve, meaning the `.catch()` in `handleLoadRed` will never run.

For consistent error handling across your API methods, it's better to re-throw the error here, just as you do in the `getAll` function.